server/src/main/java/org/cloudfoundry/identity/uaa/provider/saml/ConfiguratorRelyingPartyRegistrationRepository.java

```diff
@@ -38,19 +36,26 @@ public ConfiguratorRelyingPartyRegistrationRepository(String uaaWideSamlEntityID
     @Override
     public RelyingPartyRegistration findByRegistrationId(String registrationId) {
         IdentityZone currentZone = retrieveZone();
-        List<SamlIdentityProviderDefinition> identityProviderDefinitions = configurator.getIdentityProviderDefinitionsForZone(currentZone);
+        AbstractIdentityProviderDefinition idpDefinition = configurator.getIdentityProviderDefinitionsForOrigin(currentZone, registrationId);
+        if (idpDefinition == null) {
+            idpDefinition = configurator.getIdentityProviderDefinitionsForIssuer(currentZone, registrationId);
+        }
+        if (idpDefinition instanceof SamlIdentityProviderDefinition foundSamlIdentityProviderDefinition) {
+            return createRelyingPartyRegistration(foundSamlIdentityProviderDefinition.getIdpEntityAlias(), foundSamlIdentityProviderDefinition, currentZone);
+        }
 
+        List<SamlIdentityProviderDefinition> identityProviderDefinitions = configurator.getIdentityProviderDefinitionsForZone(currentZone);
         for (SamlIdentityProviderDefinition identityProviderDefinition : identityProviderDefinitions) {
-            if (identityProviderDefinition.getIdpEntityAlias().equals(registrationId)) {
-                return createRelyingPartyRegistration(registrationId, identityProviderDefinition, currentZone);
+            if (registrationId.equals(identityProviderDefinition.getIdpEntityAlias()) || registrationId.equals(identityProviderDefinition.getIdpEntityId())) {
+                return createRelyingPartyRegistration(identityProviderDefinition.getIdpEntityAlias(), identityProviderDefinition, currentZone);
             }
         }
 
-        if (!identityProviderDefinitions.isEmpty()) {
+        // TODO remove hack
```

Comment:
and this is the problematic part... without it breaks the SAMLIT - idpInitated SSO

Reply:
Not needed with https://github.com/cloudfoundry/uaa/pull/3136